Core/src/Timestamp.php

```diff
@@ -53,20 +54,97 @@ class Timestamp
      */
     private $nanoSeconds;
 
+    /**
+     * @var bool
+     */
+    private $fromDt = false;
```

Comment:
What about naming this `nanosFromDt`, or something similar? Otherwise it seems to say that the whole Timestamp object was from a Dt.